sky/backends/onprem_utils.py

```diff
@@ -181,20 +182,33 @@ def check_and_install_local_env(ips: List[str], auth_config: Dict[str, str]):
     def _install_and_check_dependencies(
             runner: command_runner.SSHCommandRunner) -> None:
         # Checks for python3 installation.
-        backend_utils.run_command_and_handle_ssh_failure(
+        python_version = backend_utils.run_command_and_handle_ssh_failure(
             runner, ('python3 --version'),
             failure_message=f'Python3 is not installed on {runner.ip}.')
+        python_version = python_version.split(' ')[-1].strip()
+        python_version = version.Version(python_version)
+        min_python_version = version.Version('3.6')
+        if python_version < min_python_version:
+            raise ValueError(
+                f'Python {python_version} on {runner.ip} is less than '
+                f'the minimum requirement: Python {min_python_version}.')
 
         # Checks for pip3 installation.
         backend_utils.run_command_and_handle_ssh_failure(
             runner, ('pip3 --version'),
             failure_message=f'Pip3 is not installed on {runner.ip}.')
 
         # If Ray does not exist, installs Ray.
-        backend_utils.run_command_and_handle_ssh_failure(
+        ray_version = backend_utils.run_command_and_handle_ssh_failure(
             runner, (f'ray --version || '
                      f'(pip3 install ray[default]=={sky_ray_version})'),
             failure_message=f'Ray is not installed on {runner.ip}.')
+        ray_version = ray_version.split(' ')[-1].strip()
```

Comment:
If the command above use `pip3 install`, will this `ray_version` be a different string that cannot retrieve the actual ray version with `split(' ')`?

Reply:
Good catch, fixed!

Reply:
Turns out checking `ray_version` already exists in the code. This PR will only address Python version.